cmd/soroban-cli/src/signer/keyring.rs

```diff
@@ -0,0 +1,132 @@
+use base64::{engine::general_purpose::STANDARD as base64, Engine as _};
+use ed25519_dalek::Signer;
+use keyring::Entry;
+use zeroize::Zeroize;
+
+pub(crate) const SECURE_STORE_ENTRY_PREFIX: &str = "secure_store:";
+pub(crate) const SECURE_STORE_ENTRY_SERVICE: &str = "org.stellar.cli";
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Keyring(#[from] keyring::Error),
+    #[error(transparent)]
+    Base64(#[from] base64::DecodeError),
+}
+
+pub struct StellarEntry {
+    keyring: Entry,
+}
+
+impl StellarEntry {
+    pub fn new(name: &str) -> Result<Self, Error> {
+        Ok(StellarEntry {
+            keyring: Entry::new(name, &whoami::username())?,
+        })
+    }
+
+    pub fn set_password(&self, password: &[u8]) -> Result<(), Error> {
+        let data = base64.encode(password);
+        self.keyring.set_password(&data)?;
+        Ok(())
+    }
+
+    pub fn get_password(&self) -> Result<Vec<u8>, Error> {
```

Comment:
Is this used externally? Since this will be a new public API, perhaps we should not expose the password.
```suggestion
    fn get_password(&self) -> Result<Vec<u8>, Error> {
```

Reply:
good call!